### ISSUE TITLE
Bogus warnings for comment indentation

### ISSUE THREAD

**Issue by barrywhart** [alanmcruickshank/sqlfluff#203]
Example 1:
```
SELECT
    -- Compute the thing
    (a + b) AS c
FROM
    acceptable_buckets
```

Result:
```
$ sqlfluff lint --exclude-rules L001 --dialect bigquery test.sql
== [test.sql] FAIL
L:   2 | P:   5 | L003 | Indentation not consistent with line #1
```

Example 2:
```
SELECT
    user_id
FROM
    age_data
JOIN
    audience_size
    USING (user_id, list_id)
-- We LEFT JOIN because blah
LEFT JOIN
    verts
    USING
        (user_id)
```

Output:
```
$ sqlfluff lint --exclude-rules L001 --dialect bigquery test.sql
== [test.sql] FAIL
L:   8 | P:   1 | L003 | Indentation not consistent with line #6
```

I think in both these cases, the comment should align with the code *below*, not above.

**Comment by alanmcruickshank** [comment]
Agree. Good catch.
